src/test/java/StringManipulationTest.java

```diff
@@ -50,100 +59,152 @@ public void testRemoveNthCharacter1() {
+        manipulatedstring.setString("This Test should throw IllegalArgumentException for startWord > endWord");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.getSubStrings(5, 1));
     }
+
+    // tests the case where string length is shorter than endWord and throws a
+    // IndexOutOfBoundsException if it is
     @Test
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This Test should throw IndexOutOfBoundsException for endWord > string length");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.getSubStrings(1, 100));
     }
 
     @Test
-    public void testRestoreString1()
-    {
+    public void testRestoreString1() {
         manipulatedstring.setString("art");
-        int [] array;
-        array=new int[]{1,0,2};
+        int[] array;
+        array = new int[] { 1, 0, 2 };
         String restoreString = manipulatedstring.restoreString(array);
         assertEquals(restoreString, "rat");
     }
 
+    // another test like above to check string length doesn't matter
     @Test
-    public void testRestoreString2()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString2() {
+        manipulatedstring.setString("CatDog");
+        int[] array;
+        array = new int[] { 3, 4, 5, 0, 1, 2 };
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "DogCat");
     }
 
+    // another test like above to check string length doesn't matter, even two of
+    // the same characters
     @Test
-    public void testRestoreString3()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString3() {
+        manipulatedstring.setString("backwards");
+        int[] array;
+        array = new int[] { 4, 2, 6, 7, 3, 5, 1, 0, 8 };
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "drawbacks");
     }
 
+    // tests the case where input array contains an invalid string index and throws
+    // a IndexOutOfBoundsException
     @Test
-    public void testRestoreString4()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString4() {
+        manipulatedstring.setString("wrote");
+        int[] array;
+        array = new int[] { -1, 0, 1, 2, 3 };
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.restoreString(array));
     }
 
+    // tests the case where input array is not the same length as the string and
+    // throws a IllegalArgumentException
     @Test
-    public void testRestoreString5()
-    {
-        fail("Not yet implemented");
-
+    public void testRestoreString5() {
+        manipulatedstring.setString("left");
+        int[] array;
+        array = new int[] { 0, 1, 2, 3, 4, 5 };
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.restoreString(array));
     }
 
```

Comment:
Add another test case for restoreString() for case when string contains more than 1 word, such as "Hello world".